afterpay/src/main/kotlin/com/afterpay/android/Configuration.kt

```diff
@@ -0,0 +1,26 @@
+package com.afterpay.android
+
+data class Money(
+    /**
+     * The amount is a string representation of a decimal number, rounded to 2 decimal places.
+     */
+    val amount: String,
+    /**
+     * The currency in ISO 4217 format. Supported values include "AUD", "NZD", "USD", and "CAD".
+     */
+    val currency: String
```

Comment:
I was looking at taking in strings and performing decimal conversion as well as supported currency code validation. I know kotlin doesn't have failable initialisers but that's what I was going to make use of in swift.

Reply:
I'm a bit torn on the currency code validation… is it the best experience for the merchant to be required to update the SDK to use a new region?

Reply:
You're planning on creating a failable initialiser that takes a decimal number?

That seems handy however I'm expecting that these values will in most cases be passed directly from the value returned by the `/configuration` endpoint. Do you think it's necessary to include a decimal conversion convenience initialiser?

Reply:
Hmmm, true. We might be able to validate it's supported by the framework

Reply:
The failable initialiser would take in a string and try to convert to a decimal, returning nil if it can't

Reply:
I don't expect it to fail often but we need to be able to do comparisons when creating one of the components

Reply:
Oh sure, that seems reasonable. I'll do something similar.